sensor_core/dsp/dsp_manager.py

```diff
@@ -3,59 +3,87 @@
         :param dsp_modules: Array of dsp_module objects, identified by user-defined name, target algo, and key parameters
         :param dsp_dict: master dsp_module dictionary
         :param dict_type: Dynamic dictionary """
 
         self.dsp_modules = {}
-        self.dsp_module_queue = []
+        self.dsp_modules_queue = []
 
+    @staticmethod
     def add_dsp_module(self, name: str, target_algo: str, **kwargs):
-        ma_keys = {'N': int, 'pad': str}
+        #TODO: Make sure no duplicate names, 
+        # Make sure input parameters are correct data type
+        # Currently only checks after run_dsp_module executes
+
+        ma_keys = {'window_size': int, 'pad': str}
         bw_keys = {'order': int, 'min_frq': int, 'max_frq': int, 'btype': str}
 
-        if (target_algo == 'moving_average_filter'):
+        if target_algo == 'moving_average_filter':
             if not set(ma_keys).issubset(kwargs.keys()):
                 raise ValueError(f"Missing required parameters: {ma_keys}")
             unexpected_params = set(kwargs.keys()) - set(ma_keys)
             if unexpected_params:
-                raise ValueError(f"Invalid moving average input, Unexpected parameters: {unexpected_params}")
+                raise ValueError(f"Invalid moving average input, unexpected parameters: {unexpected_params}")
             
             self.dsp_modules[name] = {'algo': target_algo, **kwargs}
-            self.dsp_module_queue.append(name)
+            self.dsp_modules_queue.append(name)
 
-        elif (target_algo == 'butterworth_filter'):
+        elif target_algo == 'butterworth_filter':
             if not set(bw_keys).issubset(kwargs.keys()):
                 raise ValueError(f"Missing required parameters: {bw_keys}")
             
             unexpected_params = set(kwargs.keys()) - set(bw_keys)
             if unexpected_params:
-                raise ValueError(f"Invalid moving average input, Unexpected parameters: {unexpected_params}")
+                raise ValueError(f"Invalid moving average input, unexpected parameters: {unexpected_params}")
             
             self.dsp_modules[name] = {'algo': target_algo, **kwargs}
-            self.dsp_module_queue.append(name)
+            self.dsp_modules_queue.append(name)
+
+    @staticmethod
+    def remove_dsp_module(self, name: str):
+        if name not in self.dsp_modules_queue:
+            raise ValueError(f"Specified module does not exist")
+        else:
+            rm_index = self.dsp_modules_queue.index(name)
+            self.dsp_modules_queue.pop(rm_index)
+            print(f"Update list: ", self.dsp_modules_queue)
+
+    def run_dsp_modules(self, data):
+        for name in self.dsp_modules_queue:
+            module = self.dsp_modules[name]
+            target_algo = module['algo']
+
+            if target_algo == 'moving_average_filter':
+                filtered_data = self.moving_average_filter(data, **module)
```

Comment:
so we need the filtered output of the previous filters to be the input for the next filter. Ideally, we should create a local copy of the input data, and overwrite that for each round of the for loop, so the overwritten copy contains the 'filtered data' and that serves as input for the next round